modules/programs/firefox.nix

```diff
@@ -21,10 +21,42 @@ let
   profilesPath =
     if isDarwin then "${firefoxConfigPath}/Profiles" else firefoxConfigPath;
 
-  nativeMessagingHostsJoined = pkgs.symlinkJoin {
-    name = "home_ff_nmhs";
-    paths = cfg.package.nativeMessagingHosts or cfg.nativeMessagingHosts;
-  };
+  nativeMessagingHosts = let
+    installNativeMessagingPath = "${mozillaConfigPath}/native-messaging-hosts";
+    sourcePackage = pkgs.symlinkJoin {
+      name = "hm_ff_nativeMessagingHosts";
+      paths = cfg.nativeMessagingHosts ++ [
+        (pkgs.runCommand "hm_ff_emptyNativeMessagingHost" { } ''
+          mkdir -p $out/lib/mozilla/native-messaging-hosts
+        '')
+      ] ++ [ cfg.finalPackage ];
```

Comment:
What is the purpose of adding this? I thought Firefox can handle both "system" and $HOME configurations of native messaging hosts

Reply:
If someone uses the NixOS implementation of native messaging hosts they'll be linked into $HOME/.mozilla automatically this way, or if they override the package with the nativemessaginghosts argument it'll be linked in as well.

Since the package is going to be referenced either way it doesn't affect anything negatively, but it enables simpler code

Reply:
The NixOS implementation of NMH doesn't work in my config, this works.

Reply:
Yes, but is this even necessary? If the native messaging hosts are already "bundled" in the final package, then I don't see why we would additionally link them in `~/.mozilla`

Reply:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests#linux